lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -425,15 +425,16 @@ def resource_object_for(serializer)
       #     id: 'required-id',
       #     meta: meta
       #   }.reject! {|_,v| v.nil? }
-      def relationships_for(serializer, requested_associations)
+      def relationships_for(serializer, requested_associations, options)
         include_tree = ActiveModel::Serializer::IncludeTree.from_include_args(requested_associations)
         serializer.associations(include_tree).each_with_object({}) do |association, hash|
           hash[association.key] = Relationship.new(
             serializer,
             association.serializer,
-            association.options,
-            association.links,
-            association.meta
+            options,
+            options: association.options,
```

Comment:
thanks for using named parameters. I love named parameters on anything with 2 or more parameters.